File: LeetCode Solutions/June 2021 LeetCoding Challenge/Maximum Units on a Truck/Maximum Units on a Truck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (a,b) -> b[1] - a[1]);
        int maxNumberOfUnits = 0;
        for (int[] b : boxTypes){
            int count = Math.min(b[0], truckSize);
            maxNumberOfUnits += count * b[1];
            truckSize -= count;
			if (truckSize == 0){
                return maxNumberOfUnits;
            }
        }
        return maxNumberOfUnits;
    }
}

File: LeetCode Solutions/LeetCoding Challenge/June 2021 LeetCoding Challenge/Maximum Units on a Truck/Maximum Units on a Truck.cpp
class Solution {
public:
    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
        sort(boxTypes.begin(), boxTypes.end(), [](auto& a, auto& b) { return b[1] < a[1];});
        int maxNumberOfUnits = 0;
        for (auto& b : boxTypes){
            int count = min(b[0], truckSize);
            maxNumberOfUnits += count * b[1], truckSize -= count;
			if (!truckSize){
                return maxNumberOfUnits;
            }
        }
        return maxNumberOfUnits;
    }
};